hydrolib/core/io/bc/models.py

```diff
@@ -136,50 +50,107 @@ def _supports_comments(cls):
     def _duplicate_keys_as_list(cls):
         return True
 
-    _header: Literal["Forcing"] = "Forcing"
+    def _convert_section_to_dict(self) -> Dict:
+        return self.dict(
+            exclude={
+                "start_line",
+                "end_line",
+            }
+        )
 
-    name: str
+    @classmethod
+    def validate(cls, v):
+        """Try to iniatialize subclass based on function field."""
+        # should be replaced by discriminated unions once merged
+        # https://github.com/samuelcolvin/pydantic/pull/2336
+        if isinstance(v, dict):
+            for c in cls.__subclasses__():
+                if (
+                    c.__fields__.get("function").default
+                    == v.get("function", "").lower()
+                ):
+                    v = c(**v)
+                    break
+        return v
+
+
+class TimeSeries(ForcingBase):
```

Comment:
I personally feel like a Forcing component has a TimeSeries function, rather than being a TimeSeries.

Reply:
While I agree the naming is super confusing between some files, these class names all derive from the section header.